### PR TITLE
add launch from python code doc

### PR DESCRIPTION
**Pull request by J-shang**
TODO: modify experiment config url

### CODE REVIEW

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
+**How to Launch from Python**
+=============================
+
+Overview
+--------
+This is a new way to launch experiments since ``nni v2.0``. Before, you need to configure our experiment in a yaml configuration file and launch the experiment through ``nnictl`` command. Now, you can also configure and run your experiment directly in the python file. If you are familiar with python programming, this will undoubtedly bring you more convenience.
```
Comment (kvartet) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:6]: our experiment -> your experiment?

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+..
+
+    Step 2 - Initialize a experiment instance and configure it
+
+.. code-block:: python
+
+    experiment = Experiment(tuner=foo_tuner, training_service='local')
+
+Now, you have a ``Experiment`` instance with ``foo_tuner`` you have initialized in the previous step, and this experiment will launch trials on your local machine due to ``training_service='local'``.
```
Comment (kvartet) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:34]: I saw `Supported value: "local", "remote", "openpai"` from the code comments, I don't know whether it supports other training services.

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+.. code-block:: python
+
+    experiment.run(port=8081)
+
+Now, you have successfully launched an NNI experiment. And you can type ``localhost:8081`` in your browser to observe your experiment in real time.
+
+Example
+-------
+Below is an example for this new launching approach. You can also find this code in ``examples/trials/mnist-tfv2/launch.py``.
```
Comment (kvartet) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:58]: maybe 
```
:githublink:`mnist-tfv2/launch.py <examples/trials/mnist-tfv2/launch.py>` 
```
will be better.

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    experiment.config.experiment_name = 'test'
+    experiment.config.trial_concurrency = 2
+    experiment.config.max_trial_number = 5
+    experiment.config.search_space = search_space
+    experiment.config.trial_command = 'python3 mnist.py'
+    experiment.config.trial_code_directory = Path(__file__).parent
+    experiment.config.training_service.use_active_gpu = True
+
+    experiment.run(8081)
+
```
Comment (kvartet) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:86]: Perhaps we should add a `note`, saying that experiments launched from python cannot be managed by nnictl or [nni client](https://nni.readthedocs.io/en/latest/nnicli_ref.html)(not sure).

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+--------
+This is a new way to launch experiments since ``nni v2.0``. Before, you need to configure your experiment in a yaml configuration file and launch the experiment through ``nnictl`` command. Now, you can also configure and run your experiment directly in the python file. If you are familiar with python programming, this will undoubtedly bring you more convenience.
+
+How to Use
+----------
+After successfully installing ``nni``, you can write your own experimental configuration and run. You can even use python script to launch multiple experiments as you like! The following three steps are necessary.
+
+..
+
+    Step 1 - Initialize a tuner you want to use
```
Comment (liuzhe-lz) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:14]: I think "step by step" is not the best way to introduce a library, seems too verbose...

Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:14]: I think it is a quick start to a new way to launch experiment, not just a library, like https://nni.readthedocs.io/en/v2.0/Tutorial/QuickStart.html#three-steps-to-start-an-experiment

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 1 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst]: fixed it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 2 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:34]: fixed it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 3 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst]: fixed it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 5 above)
Comment (liuzhe-lz) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:14]: Well I think writing Python program and using a "ctl" tool is different. But I'm okay with current version.

**Review by liuzhe-lz — APPROVED** [review]

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 6]
```diff
@@ -0,0 +1,103 @@
+**How to Launch from Python**
+=============================
+
+Overview
+--------
+This is a new way to launch experiments since ``nni v2.0``. Before, you need to configure your experiment in a yaml configuration file and launch the experiment through ``nnictl`` command. Now, you can also configure and run your experiment directly in the python file. If you are familiar with python programming, this will undoubtedly bring you more convenience.
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:6]: "the python file" -> "a python file"

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+After successfully installing ``nni``, you can write your own experimental configuration and run. You can even use python script to launch multiple experiments as you like! The following three steps are necessary.
+
+..
+
+    Step 1 - Initialize a tuner you want to use
+
+
+.. code-block:: python
+
+    from nni.algorithms.hpo.foo_tuner import FooTuner
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:19]: do we really have `FooTuner`? better to use a real and simple tuner as example

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    foo_tuner = FooTuner()
+
+Very simple, you have successfully initialized a ``FooTuner`` instance called ``foo_tuner``.
+
+See all real `builtin tuners <../builtin_tuner.rst>`__ supported in NNI.
+
+..
+
+    Step 2 - Initialize a experiment instance and configure it
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:28]: "a experiment" -> "an experiment"

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+    experiment = Experiment(tuner=foo_tuner, training_service='local')
+
+Now, you have a ``Experiment`` instance with ``foo_tuner`` you have initialized in the previous step, and this experiment will launch trials on your local machine due to ``training_service='local'``.
+
+See all `training services <../training_services.rst>`__ supported in NNI.
+
+.. code-block:: python
+
+    experiment.config.experiment_name = 'test'
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:40]: better to make this part more real, for example, setting up a real local experiment

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+Use the form like ``experiment.config.foo = 'bar'`` to configure your experiment.
+
+See `parameter configuration <TBD>`__ required by different training services.
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:44]: here is a broken link

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+    experiment.run(8081)
+
+API
+---
+
+..  autoclass:: nni.experiment.Experiment
+    :members:
+
+..  autoclass:: nni.experiment.TrialJob
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:93]: can this class be used in the new launch approach?

File: nni/experiment/experiment.py [hunk 12]
```diff
@@ -69,7 +69,7 @@ def __init__(self, tuner: Tuner, training_service: str) -> None:
             A tuner instance.
         training_service
             Name of training service.
-            Supported value: "local", "remote", "openpai".
+            Supported value: "local", "remote", "openpai", "aml", "kubeflow", "frameworkcontroller", "adl".
```
Comment (QuanluZhang) [inline nni/experiment/experiment.py:72]: so we support all the training services in v2.0?

Comment (J-shang) [reply nni/experiment/experiment.py:72]: It seems we don't support `heterogeneous` and `DLTS`? @liuzhe-lz

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 10 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:44]: waiting for #3222

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 6 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst]: fix it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 7 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst]: fix it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 8 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst]: fix it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 9 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:40]: fix it

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 11 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:99]: I will check it

File: nni/experiment/experiment.py (hunk 12 above)
Comment (QuanluZhang) [reply nni/experiment/experiment.py:72]: you can add "and hybrid training service."

**Review by QuanluZhang — APPROVED** [review]

**Review by scarlett2018 — APPROVED** [review]
I made some change suggestions directly to the Howto.rst file.

File: nni/experiment/experiment.py (hunk 12 above)
Comment (J-shang) [reply nni/experiment/experiment.py]: fix it

**Pull request merged**